cli/cli/commands/kurtosis_context/context_switch/switch.go

```diff
@@ -94,8 +100,17 @@ func run(ctx context.Context, _ *flags.ParsedFlags, args *args.ParsedArgs) error
 			return stacktrace.Propagate(err, "Error switching Kurtosis portal context")
 		}
 	}
+	logrus.Infof("Context switched to '%s', Kurtosis engine will now be restarted", contextIdentifier)
+
+	if err = engineManager.StopEngineIdempotently(ctx); err != nil {
+		return stacktrace.Propagate(err, "Unable to stop the Kurtosis engine currently running. Context will be rolled back to '%s'", contextPriorToSwitch.GetName())
+	}
+	_, _, err = engineManager.StartEngineIdempotentlyWithDefaultVersion(ctx, logrus.InfoLevel)
+	if err != nil {
+		return stacktrace.Propagate(err, "Unable to start a new Kurtosis engine. Context will be rolled back to '%s' and the Kurtosis engine will remain stopped", contextPriorToSwitch.GetName())
```

Comment:
Should we tell the user how to restart the engine?

If the user is already playing with contexts maybe they know how engine restart works so I'm fine as is

Reply:
I can add!